File: examples/platforms/android/app/src/test/java/com/here/android/test/StaticByteArrayMethodsTest.java
/*
 * Copyright (C) 2016-2018 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */
package com.here.android.test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertTrue;

import android.os.Build;
import android.support.compat.BuildConfig;
import com.here.android.RobolectricApplication;
import java.util.Arrays;
import java.util.Random;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

@RunWith(RobolectricTestRunner.class)
@Config(
  sdk = Build.VERSION_CODES.M,
  application = RobolectricApplication.class,
  constants = BuildConfig.class
)
public final class StaticByteArrayMethodsTest {
  private static final int LARGE_BYTE_ARRAY_SIZE = 100000;

  @Test
  public void returnReverseByteBuffer_smallByteArray() {
    // Arrange
    byte[] input = "InputBytes".getBytes();

    // Act
    byte[] output = StaticByteArrayMethods.returnReverseByteBuffer(input);

    // Assert
    assertEquals("setyBtupnI", new String(output));
  }

  @Test
  public void returnInputByteBuffer_emptyByteArray() {
    // Arrange
    byte[] input = {};

    // Act
    byte[] output = StaticByteArrayMethods.returnInputByteBuffer(input);

    // Assert
    assertTrue(Arrays.equals(input, output));
  }

  @Test
  public void returnInputByteBuffer_largeByteArray() {
    // Arrange
    byte[] input = new byte[LARGE_BYTE_ARRAY_SIZE];
    new Random().nextBytes(input);

    // Act
    byte[] output = StaticByteArrayMethods.returnInputByteBuffer(input);

    // Assert
    assertTrue(Arrays.equals(input, output));
  }

  @Test
  public void concatenateByteBuffers_twoSimpleByteArrays() {
    // Arrange
    byte[] input1 = "byte".getBytes();
    byte[] input2 = "array".getBytes();

    // Act
    byte[] output = StaticByteArrayMethods.concatenateByteBuffers(input1, input2);

    // Assert
    assertEquals("bytearray", new String(output));
  }
}

File: functional-tests/functional/android-kotlin/src/test/kotlin/com/here/android/test/StaticByteArrayMethodsTest.kt
/*
 * Copyright (C) 2016-2025 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */
package com.here.android.test

import com.here.android.RobolectricApplication
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

import kotlin.random.Random

@RunWith(RobolectricTestRunner::class)
@Config(application = RobolectricApplication::class)
class StaticByteArrayMethodsTest {

    @org.junit.Test
    fun returnReverseByteBuffer_smallByteArray() {
        val input: ByteArray = "InputBytes".toByteArray()
        val output: ByteArray = StaticByteArrayMethods.returnReverseByteBuffer(input)

        assertEquals("setyBtupnI", String(output))
    }

    @org.junit.Test
    fun returnInputByteBuffer_emptyByteArray() {
        val input: ByteArray = byteArrayOf()
        val output: ByteArray = StaticByteArrayMethods.returnInputByteBuffer(input)

        assertTrue(input.contentEquals(output))
    }

    @org.junit.Test
    fun returnInputByteBuffer_largeByteArray() {
        val input: ByteArray = ByteArray(100000)
        Random.nextBytes(input)

        val output: ByteArray = StaticByteArrayMethods.returnInputByteBuffer(input)
        assertTrue(input.contentEquals(output))
    }

    @org.junit.Test
    fun concatenateByteBuffers_twoSimpleByteArrays() {
        val input1: ByteArray = "byte".toByteArray()
        val input2: ByteArray = "array".toByteArray()

        val output: ByteArray = StaticByteArrayMethods.concatenateByteBuffers(input1, input2)

        assertEquals("bytearray", String(output))
    }

    @org.junit.Test
    fun reverseBlobInStruct_smallBlob() {
        val input = StaticByteArrayMethods.StructWithBlob("InputBytes".toByteArray())
        val result = StaticByteArrayMethods.reverseBlobInStruct(input)

        assertEquals("setyBtupnI", String(result.blob))
    }

    @org.junit.Test
    fun reverseBlobInStruct_emptyBlob() {
        val input = StaticByteArrayMethods.StructWithBlob(ByteArray(0))
        val result = StaticByteArrayMethods.reverseBlobInStruct(input)

        assertEquals(input.blob.size, result.blob.size)
    }
}